src/mono_font/mono_text_style.rs

```diff
@@ -0,0 +1,834 @@
+use crate::{
+    draw_target::DrawTarget,
+    geometry::{Point, Size},
+    mono_font::{MonoCharPixels, MonoFont},
+    pixelcolor::{BinaryColor, PixelColor},
+    primitives::Rectangle,
+    text::{TextMetrics, TextRenderer, VerticalAlignment},
+    Pixel, SaturatingCast,
+};
+
+/// Style properties for text using a monospaced font.
+///
+/// A `MonoTextStyle` can be applied to a [`Text`] object to define how the text is drawn.
+///
+/// Because `MonoTextStyle` has the [`non_exhaustive`] attribute, it cannot be created using a
+/// struct literal. To create a `MonoTextStyle` with a given text color and transparent
+/// background, use the [`new`] method. For more complex text styles, use the
+/// [`MonoTextStyleBuilder`].
+///
+/// [`Text`]: ../text/struct.Text.html
+/// [`non_exhaustive`]: https://blog.rust-lang.org/2019/12/19/Rust-1.40.0.html#[non_exhaustive]-structs,-enums,-and-variants
+/// [`MonoTextStyleBuilder`]: ./struct.MonoTextStyleBuilder.html
+/// [`new`]: #method.new
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+#[non_exhaustive]
+pub struct MonoTextStyle<C, F> {
+    /// Text color.
+    pub text_color: Option<C>,
+
+    /// Background color.
+    pub background_color: Option<C>,
+
+    /// Underline color.
+    pub underline_color: DecorationColor<C>,
```

Comment:
Silly question - once set, how do I turn these off?

Reply:
Actually, I don't see how I turn them on if I have a style object already. I guess [this comment](https://github.com/embedded-graphics/embedded-text/pull/90#discussion_r555892950) isn't implemented yet.

Reply:
Normally it would be possible to convert the style to a builder and set the decorations to None color, but unfortunately I don't think I can do that through the TextRenderer trait.

Reply:
You are correct that there currently is no way to set any color back to `None`. This limitation also exists in `PrimitiveStyleBuilder` and should be fixed in both (in a future PR). There was some discussion about this when we introduced `PrimitiveStyleBuilder`, but I don't remember the details and why we decided not to add any method to reset colors.

> Actually, I don't see how I turn them on if I have a style object already. I guess this comment isn't implemented yet.

I want to get this PR merged first and add additional traits in the next PRs. The suggested trait can be used by e-t or any other library, but we will need separate implementations for the user facing API. I don't want to use the same trait, because it contains methods that aren't supported by every text renderer (like `set_bold`).

> Normally it would be possible to convert the style to a builder and set the decorations to None color, but unfortunately I don't think I can do that through the TextRenderer trait.

That's the reason I want to add the `CharacterStyle` trait.

Reply:
Resetting is a must-have for SGR parameters. It's not just the colors, but the decorations themselves as well. I guess it's not an urgent blocker though, there is a lot of work still beside escape sequences.

I'm fine with the idea of an additional, dev-facing trait, I just hope I don't have to drop the fancy features :P

Reply:
> It's not just the colors, but the decorations themselves as well.

Setting `underline_color` or `strikethrough_color` is the way to disable these decorations. The `CharacterStyle` example did just use the wrong types in the setter functions.

> I guess it's not an urgent blocker though, there is a lot of work still beside escape sequences.

I'll try to work on this as soon as this PR is merged. I'm just waiting for @jamwaffles review.

> I'm fine with the idea of an additional, dev-facing trait, I just hope I don't have to drop the fancy features :P

I think this should be possible and we can still break the API until 0.7 is released.